### PR TITLE
Batch API call structure refactoring

### PR DESCRIPTION
**Pull request by jhkim1215**
## Change description

- Changed classes and functions that use the Curate batch APIs as the specs have been modified
- Instead of uploading batch job parameters to the s3, the client now directly sends the parameters via HTTP body

### CODE REVIEW

File: spb_curate/curate/api.py [hunk 1]
```diff
@@ -2257,9 +2256,9 @@ def delete_bulk(
         access_key: Optional[str] = None,
         team_name: Optional[str] = None,
         dataset_id: str,
-        images: List[Image] = [],
-        image_ids: List[str] = [],
-        image_keys: List[str] = [],
+        images: Optional[List[Image]] = None,
+        image_ids: Optional[List[str]] = None,
+        image_keys: Optional[List[str]] = None,
```
Comment (yoosful) [inline spb_curate/curate/api.py:2261]: i believe @cosseyluke deliberately kept this not optional, so plz check

File: spb_curate/curate/api.py [hunk 2]
```diff
@@ -8,25 +8,15 @@
...
-
 from spb_curate import api_requestor, error, util
-from spb_curate.abstract.api.resource import (
-    CreateResource,
-    DeleteResource,
-    ModifyResource,
-    PaginateResource,
-)
+from spb_curate.abstract.api.resource import (CreateResource, DeleteResource,
+                                              ModifyResource, PaginateResource)
```
Comment (yoosful) [inline spb_curate/curate/api.py:13]: sharing linter & formatter settings would be desirable

**Review by yoosful — APPROVED** [review]
LGTM, with minor comments. getting @cosseyluke 's review as well would be preferable

**Comment by cosseyluke** [comment]
@jhkim1215 please also update the changelog.

File: spb_curate/curate/api.py (hunk 1 above)
Comment (cosseyluke) [reply spb_curate/curate/api.py:2271]: It's handled correctly, so no problem from my side.

**Review by cosseyluke — APPROVED** [review]
Nice, good work.

**Pull request merged**